### PR TITLE
Support paste plain text with clickable links in RoosterJs 

### PR DESCRIPTION
**Pull request by gm-al**
This is a reference for the issue #1652. This PR expands on the current implementation for handling the paste functionality in RoosterJs. This supports pasting content with plain text and clickable links.

### CODE REVIEW

File: packages/roosterjs-editor-dom/lib/clipboard/handleTextPasteWithLinks.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    });
+}
+
+/**
+ * @internal
+ * Transform \t characters into EN SPACE characters
+ * @param input String NOT containing \n characters
+ * @example t("\thello", 2) => "&ensp;&ensp;&ensp;&ensp;hello"
+ */
+export function transformTabCharacters(input: string, initialOffset: number = 0) {
```
Comment (gm-al) [inline packages/roosterjs-editor-dom/lib/clipboard/handleTextPasteWithLinks.ts:62]: Would it be a good thing to move functions that are used in multiple places like `transformTabCharacters` to the utils folder and make it a common function to be used in `handleTextPasteWithClickableLinks` and `handleTextPaste`? Thank you :).

File: packages/roosterjs-content-model-types/lib/enum/PasteType.ts [hunk 2]
```diff
@@ -20,4 +20,9 @@ export type PasteType =
     /**
      * If there is a image uri in the clipboard, paste the content as image element
      */
-    | 'asImage';
+    | 'asImage'
+
+    /**
+     * If there is plain text with links in the clipboard, paste the content as plain text with clickable links
+     */
```
Comment (JiuqingSong) [inline packages/roosterjs-content-model-types/lib/enum/PasteType.ts:27]: Ok I think now I understand what you mean. You want to paste as plain text to remove all formats, but only keep links?

I don't know if it is a good idea. But I feel this should be done in your own code, but not as part of roosterjs.

Comment (gm-al) [reply packages/roosterjs-content-model-types/lib/enum/PasteType.ts:27]: Yes, I thought to help submit a PR to add a new option where links can appear in plain text/clickable. If you don't think this is a good idea to continue, I can close this PR?

I thought when you mentioned in the issue that he could write his own plugin and contribute back- this option could be a good thing to have in roosterjs  :).

Comment (JiuqingSong) [reply packages/roosterjs-content-model-types/lib/enum/PasteType.ts:27]: please feel free to close

**Pull request closed without merging**